src/admin/Guardian.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {Root} from "src/Root.sol";
+
+interface SafeLike {
+    function getOwners() external view returns (address[] memory);
+    function isOwner(address signer) external view returns (bool);
+    function getThreshold() external view returns (uint256);
+}
+
+interface RouterAggregatorLike {
+    function disputeMessageRecovery(bytes32 messageHash) external;
+}
+
+/// @title  Guardian
+/// @dev    This contract allows a Gnosis Safe to schedule and cancel new relys,
+///         and unpause the protocol through the timelock of Root. Additionally,
+///         it allows any owners of the safe to instantly pause the protocol.
+
+contract Guardian {
+    Root public immutable root;
+    SafeLike public immutable safe;
+    RouterAggregatorLike public immutable aggregator;
+
+    constructor(address root_, address safe_, address aggregator_) {
```

Comment:
I'd swap this, as root and aggregator are the external dependencies so should fit together
```suggestion
    constructor(address safe_, address root_, address aggregator_) {
```